fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintenanceTest.java

```diff
@@ -403,6 +434,314 @@ void flakyMerge(boolean isGrouped,
+        // save a record
+        try (FDBRecordContext context = openContext(contextProps)) {
+            Pair<FDBRecordStore, QueryPlanner> storeAndPlanner = schemaSetup.apply(context);
+
+            TestRecordsTextProto.ComplexDocument cd = TestRecordsTextProto.ComplexDocument.newBuilder()
+                    .setGroup(1)
+                    .setDocId(1000L)
+                    .setIsSeen(true)
+                    .setText("A word about what I want to say")
+                    .setTimestamp(timestamp)
+                    .setHeader(TestRecordsTextProto.ComplexDocument.Header.newBuilder().setHeaderId(999L))
+                    .build();
+            Tuple primaryKey = storeAndPlanner.getLeft().saveRecord(cd).getPrimaryKey();
+            insertedDocs.computeIfAbsent(Tuple.from(1), k -> new HashMap<>()).put(primaryKey, Tuple.from(timestamp));
+            commit(context);
+        }
+
+        // explicitly lock directory then commit
+        try (FDBRecordContext context = openContext(contextProps)) {
+            Pair<FDBRecordStore, QueryPlanner> storeQueryPlannerPair = schemaSetup.apply(context);
+            LuceneIndexMaintainer luceneIndexMaintainer = (LuceneIndexMaintainer)storeQueryPlannerPair.getLeft().getIndexMaintainer(index);
+            FDBDirectory fdbDirectory = luceneIndexMaintainer.getDirectory(Tuple.from(1), 0);
+            Lock testLock = fdbDirectory.obtainLock("write.lock");
+            testLock.ensureValid();
+            commit(context);
+        }
+
+        // create another record
+        try (FDBRecordContext context = openContext(contextProps)) {
+            Pair<FDBRecordStore, QueryPlanner> storeQueryPlannerPair = schemaSetup.apply(context);
+
+            TestRecordsTextProto.ComplexDocument cd = TestRecordsTextProto.ComplexDocument.newBuilder()
+                    .setGroup(1)
+                    .setDocId(2000L)
+                    .setIsSeen(true)
+                    .setText("A word about what I want to say")
+                    .setTimestamp(timestamp + 2000)
+                    .setHeader(TestRecordsTextProto.ComplexDocument.Header.newBuilder().setHeaderId(1999L))
+                    .build();
+            Tuple primaryKey = storeQueryPlannerPair.getLeft().saveRecord(cd).getPrimaryKey();
+            insertedDocs.computeIfAbsent(Tuple.from(1), k -> new HashMap<>()).put(primaryKey, Tuple.from(timestamp + 2000));
+            commit(context);
+        }
+
+        // validate index
+        new LuceneIndexTestValidator(() -> openContext(contextProps), context -> {
+            Pair<FDBRecordStore, QueryPlanner> storeQueryPlannerPair = schemaSetup.apply(context);
+            return storeQueryPlannerPair.getLeft();
+        }).validate(index, insertedDocs, Integer.MAX_VALUE, "text:about", false);
+    }
+
+
+    // A chaos test of two threads, one constantly trying to merge, and one trying to update a record, or save a new record or do a search.
```

Comment:
Here also, we're not doing a search, but testing that we can would be good. I think searches should succeed when there is a merge ongoing, but updates/saves should not.

Reply:
I would be in support of just updating the comment in this PR, and then following up with additional tests